### ISSUE TITLE
I'll work on Select the 9 groups (open circles)

### ISSUE THREAD

**Issue by laserlab** [ubsuny/CP1-24-HW4#8]
I'll work on 

- Select the 9 groups (open circles) in Figure 1 of [Hubble's 1929 article](https://www.pnas.org/content/pnas/15/3/168.full.pdf)(due till Monday 2pm)(1 member)

_Originally posted by @iglesias-cardinale in https://github.com/ubsuny/CP1-24-HW4/issues/1#issuecomment-2439700194_

**Comment by iglesias-cardinale** [comment]
@Priyanka-43 I started to write something to extract the data from the paper. I just copy and pasted the data into a markdown file called "raw_data.md" in my repository and formatted it a little bit. I was (almost) able to reproduce the plot that is in the paper. I say almost because a couple of the data points are in different spots. I think it's because I used the data from table 1 instead of table 2. 

I'm still trying to figure out exactly how the data is grouped, but we now have access to the data. I'm not sure of the best way to share the data file and short bit of code that I wrote with you.

**Comment by iglesias-cardinale** [comment]
I plotted all the data from Table two and that actually did not fix the problem. I think the actual problem is that Hubble's plot was made by hand and Python does a better job of plotting the data points accurately. 

I also haven't been able to figure out exactly how the data was grouped. In the paper, Hubble mentions that the data groups are selected based on their proximity both in distance and in velocity, but he doesn't give a clear/quantifiable method for the selection. I tried dividing both the velocity and distance ranges into nine equal sub-ranges, sorting the data into each of those sub-ranges and averaging the sorted data within each sub-range, but the results didn't match up with the plot in the paper at all. The data, averaged in this way, was much more linear than that in the paper.

**Comment by laserlab** [comment]
> I plotted all the data from Table two and that actually did not fix the problem. I think the actual problem is that Hubble's plot was made by hand and Python does a better job of plotting the data points accurately.
> 
> I also haven't been able to figure out exactly how the data was grouped. In the paper, Hubble mentions that the data groups are selected based on their proximity both in distance and in velocity, but he doesn't give a clear/quantifiable method for the selection. I tried dividing both the velocity and distance ranges into nine equal sub-ranges, sorting the data into each of those sub-ranges and averaging the sorted data within each sub-range, but the results didn't match up with the plot in the paper at all. The data, averaged in this way, was much more linear than that in the paper.

It’s good you opened a PR so it can be reviewed and merged in time. If you cannot exactly reproduce Hubble plot it’s fine as long as you coordinate with @Priyanka-43 to write about it in the documentation.

**Comment by iglesias-cardinale** [comment]
I actually think I figured it out. I mentioned it in PR #16 but I forgot to mention it here. I ended up grouping the data by the first column in the table. 

The first column is a bunch of numerical values, but they correspond to the names/labels of nebulae in the NGC Catalog. NGC labels nebulae that are close to one another with similar numerical values. The labels range in values from ~200 - ~7000, so I grouped in such a way that all nebulae with labels between 0 and 1000 were in one group, all nebulae with labels between 1000 and 2000 were in another group, and so on.
